### PR TITLE
fix: Profile page links shows background color on hover

### LINKED ISSUE #7042: [BUG] Profile page links on hover are not accessible
### Description

Current focus indicate for links is changing the border, but it would be good to change the background.

UPDATE...

On hover (especially in dark mode) only the border of the link changes colour, the background should change also to show it is being hovered

- the background colour change on hover will be needed for light and dark mode
- we have prebuilt classes you can use from `tailwind.config.js`

Any questions let us know

Screenshot 2023-05-26 at 20 15 36

### Screenshots

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by yashug**
## Fixes Issue

Closes #7042  -->

## Changes proposed

- [X] - Added background color on hover for light theme
- [X] - Added background color on hover for dark theme

## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

## Screenshots

Screenshot 2023-06-13 at 5 59 48 PM
Screenshot 2023-06-13 at 6 00 00 PM

## Note to reviewers

Could you please suggest an alternative background color for both light and dark themes? I have currently used a secondary-low color, as I find it more prominent than other colors. I would appreciate any suggestions you may have. Thank you!

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/7556"></a>

### CODE REVIEW

**Review by dan-mba — CHANGES_REQUESTED** [review]

File: components/user/UserLink.js [hunk 1]
```diff
@@ -19,7 +19,7 @@ export default function UserLink({ BASE_URL, link, username }) {
       )}`}
       target="_blank"
       rel="noopener noreferrer"
-      className="rounded-full border border-primary-medium-low dark:border-primary-medium-low dark:hover:border-[color:var(--hover-color)] hover:border-[color:var(--hover-color)] hover:shadow-xl p-4 my-2 w-full content-start flex flex-row gap-4 items-center dark:bg-primary-medium"
+      className="rounded-full border border-primary-medium-low dark:border-primary-medium-low dark:hover:border-[color:var(--hover-color)] hover:border-[color:var(--hover-color)] hover:shadow-xl p-4 my-2 w-full content-start flex flex-row gap-4 items-center dark:bg-primary-medium dark:hover:bg-secondary-low hover:bg-secondary-low"
```
[the remark this answers is not in the document]
Comment (yashug) [reply components/user/UserLink.js]: @dan-mba changed the contrast from 1.99 to 7.1 in dark mode

Screenshot 2023-06-13 at 7 45 42 PM

**Review by dan-mba — DISMISSED** [review]
The change meets the accessibility guidelines.
As UI outside my area of expertise, I will let another maintainer approve your color choices.

File: components/user/UserLink.js [hunk 2]
```diff
@@ -19,7 +19,7 @@ export default function UserLink({ BASE_URL, link, username }) {
       )}`}
       target="_blank"
       rel="noopener noreferrer"
-      className="rounded-full border border-primary-medium-low dark:border-primary-medium-low dark:hover:border-[color:var(--hover-color)] hover:border-[color:var(--hover-color)] hover:shadow-xl p-4 my-2 w-full content-start flex flex-row gap-4 items-center dark:bg-primary-medium"
+      className="rounded-full border border-primary-medium-low dark:border-primary-medium-low dark:hover:border-[color:var(--hover-color)] hover:border-[color:var(--hover-color)] hover:shadow-xl p-4 my-2 w-full content-start flex flex-row gap-4 items-center dark:bg-primary-medium dark:hover:bg-secondary-low/40 hover:bg-secondary-low"
```
Comment (eddiejaoude) [inline components/user/UserLink.js:22]: I feel `/40` for the light mode hover looks better also - we will update these later with the full rebrand.

```suggestion
      className="rounded-full border border-primary-medium-low dark:border-primary-medium-low dark:hover:border-[color:var(--hover-color)] hover:border-[color:var(--hover-color)] hover:shadow-xl p-4 my-2 w-full content-start flex flex-row gap-4 items-center dark:bg-primary-medium dark:hover:bg-secondary-low/40 hover:bg-secondary-low/40"
```

Screenshot 2023-06-14 at 10 44 50

**Comment by eddiejaoude** [comment]
Screenshot 2023-06-14 at 10 44 50

**Comment by eddiejaoude** [comment]
Great collaboration 👍

**Review by eddiejaoude — APPROVED** [review]
Thank you 👍

**Pull request merged**